libs/watt/src/lib/components/drawer/watt-drawer.component.ts

```diff
@@ -57,28 +57,48 @@ export type WattDrawerSize = 'small' | 'normal' | 'large';
   imports: [A11yModule, MatSidenavModule, WattButtonComponent, WattSpinnerComponent],
 })
 export class WattDrawerComponent implements OnDestroy {
-  private cdr = inject(ChangeDetectorRef);
   private elementRef = inject(ElementRef);
   private overlayContainer = inject(OverlayContainer);
-  private cssCustomPropertiesService = inject(WattCssCustomPropertiesService);
+  private cdkTrapFocus = viewChild.required(CdkTrapFocus);
+  private bypassClickCheck = false;
+  private writableIsOpen = signal(false);
+
+  // Multiple drawers open at the same time is not allowed. This keeps track of
+  // the currently opened drawer and closes it when a new drawer is opened.
   private static currentDrawer?: WattDrawerComponent;
 
   /** Used to adjust drawer size to best fit the content. */
   size = input<WattDrawerSize>('normal');
 
-  /** Whether the drawer is open.  */
-  isOpen = model(false);
+  /**
+   * Whether the drawer should open automatically. If `key` is provided and
+   * `autoOpen` is true, the drawer will open every time the key changes.
+   */
+  autoOpen = input(false, { transform: booleanAttribute });
 
-  /** Whether the drawer should show a loading state.  */
+  /**
+   * Used to track the current drawer when reusing the same drawer instance to
+   * render different content. This is required when interactions outside the
+   * drawer should result in updating the drawer's content instead of closing it.
+   */
+  // Transform input to a string (omitting input should not be the same as passing undefined)
+  key = input<unknown>();
```

Comment:
question: is there supposed to be a `transform` here? The comment sounds as if there is.

Reply:
Oh yeah good catch, I removed that. Turns out it wasn't needed.

da81af762588ea1f30f10a255623f7fb6c4bc454